src/helpers/InitHelper.ts

```diff
@@ -45,26 +45,30 @@ export default class InitHelper {
     // Always check for an updated service worker
     await OneSignal.context.serviceWorkerManager.updateWorker();
 
-    OneSignal.context.sessionManager.incrementPageViewCount();
-
+    OneSignal.context.pageViewManager.incrementPageViewCount();
+    Log.warn("document.visibilityState", document.visibilityState);
     if (document.visibilityState !== 'visible') {
-      once(
-        document,
-        'visibilitychange',
-        (_: any, destroyEventListener: Function) => {
-          if (document.visibilityState === 'visible') {
-            destroyEventListener();
-            InitHelper.sessionInit();
-          }
-        },
-        true
-      );
+      InitHelper.postponeSesstionInitUntilPageIsInFocus();
       return;
     }
 
     await InitHelper.sessionInit();
   }
 
+  public static postponeSesstionInitUntilPageIsInFocus(): void {
```

Comment:
Given that it does not have effect on the compiled code and since we often need access to methods in tests, I gave up on TS `private` for the most part. I started dropping `public` too since it's the default anyway. I think it would be better to introduce smaller modules and have clear directions on when to use what type of file, e.g. util, helper, manager, etc. It would benefit SDK readability and maintainability in the long run. Let's brainstorm more when you get back!